docs: initial draft for Update 67 the Tari 2021 wrap up (#99)
tari-project/tari-dot-com: _updates/2021-12-21-update-67.md

---
layout: update
tag: Developer Update
date: 2021-12-21
author: Philip Robinson
thumbnail: update-background.jpg
title: Tari 2021 wrap up
class: subpage
---

> This year has seen the slow and steady journey towards Mainnet continue. Many hard lessons have been learnt from our various 
> testnets and turned into massive stability and performance gains on the base layer. The latest testnet, Weatherwax, was 
> launched in July and since then has seen 400k transactions and contains 1.8 million UTXO’s. We also broke ground on the [DAN] 
> building the base layer features required to support the second layer and made significant inroads into the second layer 
> implementation.

Reviewer: ```suggestion
2021 is largely in the rear-view mirror. While the degens were swapping enough ETH to put small countries out of the red for pixelated animal JPEGs, the Tari devs were grinding away.

Top-level highlights include:

* TariScript - bringing Bitcoin script-like functionality to Mimblewimble.
* SHA-3 and RandomX Mining pools.
* Cross-chain atomic swaps
* One-sided payments
* Cursory DAN support
  * Hotstuff BFT consensus
  * NFT-like contracts proof of concept
* Yuge performance improvements

The Weatherwax testnet launched in July. In 6 months, it has seen 400k transactions and contains 1.8 million UTXO’s.  Because we still use Monopoly money, we can really hammer the network and test its performance. Just FYI, Grin had around 150k transactions in that time and has around 150k UTXOs. 
```

transaction and block broadcast and propagating strategies.

This year in July, we ran the same scale test and achieved a 100% success rate with all the transactions being mined within 
500 blocks where the absolute minimum number of blocks they could have fitted in was 54 blocks. Finally, in December, the 
same stress test was again 100% successful, and all the transactions were mined within 300 blocks, with 97% of them mined 
within 100 blocks from the beginning of the test. There is still work to be done, but this considerable improvement in 
stability and performance has resulted from many hard-won lessons.

For example, this year, we moved most of our network communications to an RPC-style request-response framework. This 
replaces the fire-and-forget style messaging we initially built, which is now reserved only for the sending of asynchronous 
transaction negotiation messages. There have also been many improvements to our higher-level communication strategies. 
For example, the method by which wallets broadcast transactions into the mempool, the way base nodes sync headers and blocks 
between themselves, to name just a few.

## Tari Script

> One of the most significant additions to the base layer has been the implementation of [Tari Script]. This is one of Tari’s 

Reviewer: ```suggestion
One of the most significant additions to the base layer has been the implementation of [TariScript]. This is one of Tari’s 
```

extensions to [Mimblewimble] protocol and is a scripting language that massively expands the capabilities of the Tari 
base layer. To our knowledge, none of the existing [Mimblewimble] projects has ambitions to add a scripting language to 
their protocols.

This year we rolled out the first feature to make use of [Tari Script] which was the implementation of [one-sided payments]. 
One-sided payments allow a wallet to make a payment to a recipient without the recipient’s wallet needing to be online, 
solving a long time criticism levelled at Mimblewimble based protocols. We also used [Tari Script] to implement cross-chain 
atomic swaps with [Bitcoin] and wrote an RFC for our implementation of an atomic swap with [Monero].

This is just the tip of the iceberg of the capabilities Tari Script adds to the protocol, and in 2022 we will see many 

> more applications of [Tari Script] to the base layer and the [DAN].

Reviewer: ```suggestion
more applications of [TariScript] to the base layer and the [DAN].
```

## Wallets

> The common wallet backend library used by the desktop console wallet and Android and iOS Aurora app was the focus of much 

Reviewer: ```suggestion
The common wallet backend library used by the desktop console wallet and Android and iOS Aurora app was the focus of many 
```

to scan the blocks since the birthday rather than all the way from the genesis block.

This year we created the standalone desktop console wallet client and iterated on our iOS and Android Aurora apps to make 
use of all the new features afforded by the backend, such as wallet encryption and custom base node support. Furthermore, 
the Aurora apps have been integrated with [Yat] to allow you to send tXTR to Yat’s with associated addresses in addition 
to the existing Emoji ID addresses.

## Digital Assets Network (DAN)
As the base layer stabilises, the development of the second layer [DAN] is going full steam ahead. We have implemented a 
proof of concept Validator Node application that includes a HotStuff BFT consensus algorithm and NFT-like contracts.

## Mining

This year we rolled out mining node applications for RandomX merge-mining and SHA-3 mining. The merge-mining proxy 
application enabled the merge mining of Tari along with Monero. Furthermore, we forked Nodejspool to support Tari merge 
mining and added SHA-3 mining to our fork of the Miningcore mining pool framework.

[DAN]: https://rfc.tari.com/RFC-0300_DAN.html

> [Tari Script]: https://rfc.tari.com/RFC-0201_TariScript.html

Reviewer: ```suggestion
[TariScript]: https://rfc.tari.com/RFC-0201_TariScript.html
```
